test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/CharacterTest.scala

```diff
@@ -434,9 +434,136 @@ class CharacterTest {
     assertEquals(0xff3b, Character.toLowerCase(0xff3b)) // ［ => ［
   }
 
-  @Test def toUpperCase_compare_char_and_codepoint(): Unit = {
+  @Test def toTitleCase_compare_char_and_codepoint(): Unit = {
     for (ch <- Character.MIN_VALUE to Character.MAX_VALUE)
-      assertEquals(Character.toUpperCase(ch), Character.toUpperCase(ch.toInt).toChar)
+      assertEquals(Character.toTitleCase(ch), Character.toTitleCase(ch.toInt).toChar)
+  }
+
+/*
+for (ch <- 0.toChar until Character.MAX_VALUE) {
+  val titleCase: Char = Character.toTitleCase(ch)
+  val upperCase: Char = Character.toUpperCase(ch)
+
+  if (titleCase != upperCase)
+    println(s"    assertEquals('$titleCase', Character.toTitleCase('$ch'))")
+}
+*/
+  @Test def toTitleCase_Char_UpperCase_Differences(): Unit = {
+    assertEquals('ǅ', Character.toTitleCase('Ǆ'))
+    assertEquals('ǅ', Character.toTitleCase('ǅ'))
+    assertEquals('ǅ', Character.toTitleCase('ǆ'))
+    assertEquals('ǈ', Character.toTitleCase('Ǉ'))
+    assertEquals('ǈ', Character.toTitleCase('ǈ'))
+    assertEquals('ǈ', Character.toTitleCase('ǉ'))
+    assertEquals('ǋ', Character.toTitleCase('Ǌ'))
+    assertEquals('ǋ', Character.toTitleCase('ǋ'))
+    assertEquals('ǋ', Character.toTitleCase('ǌ'))
+    assertEquals('ǲ', Character.toTitleCase('Ǳ'))
+    assertEquals('ǲ', Character.toTitleCase('ǲ'))
+    assertEquals('ǲ', Character.toTitleCase('ǳ'))
+  }
+
+/*
+def format(codePoint: Int): String = "0x%04x".format(codePoint)
+
+def myToUpperCase(cp: Int): Int = {
+  val upperString = new String(Array(cp),0,1).toUpperCase()
+  if (upperString.length() == 1)
+    upperString.charAt(0)
```

Comment:
Should this be expressed in terms of code points?
```suggestion
  if (upperString.codePointCount(0, upperString.length()) == 1)
    upperString.charPointAt(0)
```
In fact, I don't think I understand what codepoints are selected by this script and why?

Reply:
I updated this back to a previous version I had that is hopefully more straightforward to understand what is being tested.